### PR TITLE
Feature/network graph update

### PR DESCRIPTION
**Pull request by Bluesmile82**
- Add Initial letters to graph links
- Refactor graph code to be able to correctly display the svg order
- Make the items title clickable and act as a link to the corresponding network
https://vizzuality.atlassian.net/jira/software/c/projects/ORC/boards/84?assignee=5f3ceebffdc3f5003fafce9e&selectedIssue=ORC-433

### CODE REVIEW

File: client/src/components/network-diagram/index.tsx [hunk 1]
```diff
@@ -66,64 +61,86 @@ const NetworkDiagram = ({
...
+            name={name}
+            id={id}
+            type={type}
+            hasChildren={networks?.length > 0}
+            hasDot={networks?.length > 0}
+            className="z-40"
+          />
+          {
+            networks
+              .reverse() // We reverse the array to correct the z-index of the svgs
```
Comment (clementprdhomme) [inline client/src/components/network-diagram/index.tsx:74]: The `reverse`  function mutates the array. Use `[...network].reverse()` instead.

File: client/src/components/network-diagram/index.tsx [hunk 2]
```diff
@@ -66,64 +61,86 @@ const NetworkDiagram = ({
...
+              .map((network, childIndex) => {
+                if (
+                  !network ||
+                  typeof network === 'undefined' ||
```
Comment (clementprdhomme) [inline client/src/components/network-diagram/index.tsx:78]: I think this line is unnecessary as `!network` would already be `true`.

File: client/src/components/network-diagram/item.tsx [hunk 3]
```diff
@@ -91,45 +129,48 @@ const Item = ({
...
           heightIndex={heightIndex}
           category={category}
           isGranchild={isGranchild}
-          isFirstOfType={isFirstOfType}
+          hasDot={(isFirstNode && hasChildren) || (hasChildren && category && opened)}
+        />
+      )}
+      {hasDot && (
+        <span
+          className="dotss absolute left-[14px] z-20 h-2 w-2 rounded-full bg-black"
```
Comment (clementprdhomme) [inline client/src/components/network-diagram/item.tsx:149]: Is the class `dotss` intentional?

**Review by clementprdhomme — COMMENTED** [review]
An additional comment: do you think it's possible to have the coordinator link always on top?

The coordinator link is partially hidden by a partner link

**Comment by clementprdhomme** [comment]
I'm still seeing the coordinator's link below the partner's. Is that because you weren't able to display it on top?

**Comment by Bluesmile82** [comment]
It should be fixed now

**Comment by clementprdhomme** [comment]
I'm not sure the logic is fully correct. Here you stack them on top of each other based on their order. I think that you should stack them based on which type of link: coordinator > funder > partner. Do you think it's possible?

I'm giving you an example where the coordinator link is still below a partner link:
Vizzuality's graph

**Comment by Bluesmile82** [comment]
Thanks for checking! I actually forgot to apply the logic to the grandchild. Now is working. I think the logic should be as it is. The links above should always be on top of the ones below regarding their type.

**Pull request merged**